orca-keel/src/test/java/com/netflix/spinnaker/orca/keel/task/ImportDeliveryConfigTaskTest.java

```diff
@@ -0,0 +1,251 @@
+            (String) contextMap.get("directory"),
+            (String) contextMap.get("manifest"),
+            (String) contextMap.get("ref")))
+        .thenReturn(mockResponseBody);
+
+    var result = importDeliveryConfigTask.execute(stage);
+
+    assertThat(result).isEqualTo(terminal);
+  }
+
+  /**
+   * This test is a negative case which verifies if the task returns {@link
+   * ImportDeliveryConfigTask.SpringHttpError} on 4xx http error. Here the error body is mocked with
+   * timestamp in {@link ChronoUnit#NANOS}, which WILL NOT be parsed to exact timestamp in the
+   * method @see {@link ImportDeliveryConfigTask#handleRetryableFailures(RetrofitError,
+   * ImportDeliveryConfigTask.ImportDeliveryConfigContext)} and results in will contain incorrect
+   * timestamp.
+   *
+   * <p>Other cases where the timestamp will result in incorrect value, are when the units in :
+   * {@link ChronoUnit#MICROS}
+   */
+  @Test
+  public void verifyTaskResultReturnsSpringHttpErrorWhenTimestampIsInNanos()
+      throws JsonProcessingException {
+
+    var mockResponseBody =
+        Map.of(
+            "name",
+            "keeldemo-manifest",
+            "application",
+            "keeldemo",
+            "artifacts",
+            Collections.emptySet(),
+            "environments",
+            Collections.emptySet());
+
+    var httpStatus = HttpStatus.BAD_REQUEST;
+    var timestamp = Instant.now().truncatedTo(ChronoUnit.NANOS);
+
+    // Initialize SpringHttpError with timestamp in nanos and HttpStatus 400 bad request.
+    var httpError = mockSpringHttpError(httpStatus, timestamp);
+
+    var context = Map.of("error", httpError);
+
+    TaskResult terminal = TaskResult.builder(ExecutionStatus.TERMINAL).context(context).build();
+
+    // simulate 400 bad request http error with SpringHttpError body
+    simulateFault("/delivery-configs/", objectMapper.writeValueAsString(httpError), httpStatus);
+
+    when(scmService.getDeliveryConfigManifest(
+            (String) contextMap.get("repoType"),
+            (String) contextMap.get("projectKey"),
+            (String) contextMap.get("repositorySlug"),
+            (String) contextMap.get("directory"),
+            (String) contextMap.get("manifest"),
+            (String) contextMap.get("ref")))
+        .thenReturn(mockResponseBody);
+
+    var result = importDeliveryConfigTask.execute(stage);
+
+    var httpErrorBody = (ImportDeliveryConfigTask.SpringHttpError) result.getContext().get("error");
+
+    // assert all the values in the http error body are true except the timestamp
+    assertThat(httpErrorBody.getStatus()).isEqualTo(httpStatus.value());
+    assertThat(httpErrorBody.getError()).isEqualTo(httpStatus.getReasonPhrase());
+    assertThat(httpErrorBody.getMessage()).isEqualTo(httpStatus.name());
+    assertThat(httpErrorBody.getDetails()).isEqualTo(Map.of("exception", "Http Error occured"));
+    assertThat(httpErrorBody.getTimestamp()).isNotEqualTo(timestamp);
```

Comment:
Is there something more we know about httpErrorBody.getTimestamp()?  It's not equal to timestamp, but what is it?